### ISSUE TITLE
Add gapless playback

### ISSUE THREAD

**Issue by unraun** [mopidy/mopidy-spotify#160]
Especially when playing back live-albums on Spotify I can recognize small pauses between the tracks. Would be great, if that issue could be fixed.

From my experience MPD has the ability for gapless playback (buffering). Don't know if this is an issue for mopidy or mopidy-spotify.

**Comment by Soaku** [comment]
Gapless playback support was added to Mopidy in 2.0, the lastest is >3.0. I just tested, it does work for local files — but not for Spotify. This issue is already 3 years old, yet this is a must-have!

**Comment by Soaku** [comment]
Mopidy applies gapless playback automatically, and so, it isn't an issue with `mopidy-spotify`, even though you won't encounter this with other extensions like `mopidy-file`, `mopidy-soundcloud` or `mopidy-gmusic`. The reason it, unlike them, `mopidy-spotify` sends audio to mopidy using appsrc with seems to be bugged when it comes to switching songs — some weird data is sent to its buffer, even after the song has ended. I'll open an issue in Mopidy, or attempt to fix this myself.

But, in the meantime, here's a workaround:

`mopidy/audio/actor.py`, [line 97](https://github.com/mopidy/mopidy/blob/develop/mopidy/audio/actor.py#L97), put this below it:

```py
if buffer_.duration == 500_000_000:
    return True
result = self._source.emit("push-buffer", buffer_)
return result == Gst.FlowReturn.OK
```

**Comment by Sirnut** [comment]
I know this issue has been closed, but I did want to report something experienced earlier

I set up a fresh install of Pirate Audio on the line out board for a zero WH [using the guide here as reference.](https://learn.pimoroni.com/tutorial/sandyj/getting-started-with-pirate-audio)

I'm not sure why but initially gapless playback was not working properly on my device. I messed around with stuff and eventually ran the command:  `sudo pip3 install --upgrade mopidy-spotify` 

Afterwards gapless playback is working nicely, although I'm not sure why a fresh install of the software would not have already included the updates found in the install script.

I just wanted to mention this in case it is important in the future!
